routes/Favorite.js

```diff
@@ -0,0 +1,42 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const favorite = require('../db/favorite');
+
+const router = express.Router();
+var urlencodedParser = bodyParser.urlencoded({extended: false});
+
+router.post('/addFav', urlencodedParser, async(req, res) => {
+    let username = await favorite.Favorite.findOne({
+        username: req.body.username
+    });
+
+    if (!username) {
+        var newUser = new favorite.Favorite(req.body);
+        newUser.save().then(item => {
+            res.send('added to favorite')
+        }) 
+    } else {
+        // if user have been used favorite before then update the list 
+        var userfav = 0;
+        username.favorite.forEach( function(item, index) {
+            if (item === req.body.favorite) userfav = 1;
+        })
+
+        if (!userfav) {
+            var newVal = { $push: {favorite: req.body.favorite} };
```

Comment:
This seems to be removing all the previous one.
what is in req.body.favorite?

Reply:
celebrity name like BTS, TWICE,

Reply:
if it is list, it should be fine.
just double check.